### ISSUE TITLE
Blazor support of task based parallelism.

### ISSUE THREAD

**Issue by RChrisCoble** [dotnet/runtime#40619]
Is it in the roadmap for Blazor to support .Net's task based parallelism with WASM threads in .Net 5 or 6?

For instance, Chrome released WASM threads support in version 70.  (I'm running version 84), so it's been out for some time.  I'm not looking for async/await, I'm looking for maximizing the core usage through Task().

**Comment by danroth27** [comment]
I believe the main limitation here is browser support for threads. While Chromium based browsers and Firefox I believe now have threading support on desktop, I believe Safari still doesn't and most mobile browser also still don't: https://caniuse.com/?search=sharedarraybuffer.

@lewing Can you confirm?

**Comment by RChrisCoble** [comment]
The Uno team seems to be messing around in this space already.

https://platform.uno/blog/webassembly-threading-in-net/

**Comment by danroth27** [comment]
@RChrisCoble Yup, that's all based on the work we've done in the .NET WebAssembly runtime to support threads. It just has limited browser applicability currently.

**Comment by RChrisCoble** [comment]
@danroth27 thanks for the reply.  So to put a fine point on the current status of task based parallelism using browser thread support:  (cc'ing @lewing )

1.  The combined .Net 5/Blazor release will not support WASM task based parallelism, and everything will continue to run single threaded like the current Blazor bits (and JS).
2. There is work in flight to support WASM task based parallelism in the .Net core runtime.  Does this have a release vehicle?  .Net 6?
3. Even with the support of #2 above, it will only work in browsers that support threads.

**Comment by RChrisCoble** [comment]
For context on what we're trying to do.  We want to take the .Net engine that drives this:

https://www.aveva.com/content/dam/aveva/images/cards/664x498/W009_ADNOC_10-20.jpg.rendition/mobile-1x.png

And run the exact same code in a browser using Blazor.  It would be quite the success story for the power of Blazor, in my opinion.  We also need .Net AOT compilation so we can draw with SkiaSharp.  @mattleibow sounds like he's ready to knock that out when .Net 6 preview releases of AOT start dropping.

Then take that same work and drop it on mobile with MBB, and/or desktop with the features I now see slated for release with .Net 6/ASP.Net.  

You guys are doing some extremely exciting work.  Next year can't come fast enough.

**Comment by danroth27** [comment]
> There is work in flight to support WASM task based parallelism in the .Net core runtime. Does this have a release vehicle? .Net 6?

We're still in the process of planning for .NET 6, but given the state of browser support for threading and the amount of runtime work we already want to do for .NET 6 I'd say it's looking unlikely that multithreading will land for Blazor WASM in .NET 6.

> We also need .Net AOT compilation so we can draw with SkiaSharp. @mattleibow sounds like he's ready to knock that out when .Net 6 preview releases of AOT start dropping.

I do expect that WebAssembly AoT compilation will land for .NET 6, and the current thinking is that this will enable SkiaSharp scenarios. Still lots to work out though - .NET 6 planning is still a work in progress.

**Comment by jtotec** [comment]
> > There is work in flight to support WASM task based parallelism in the .Net core runtime. Does this have a release vehicle? .Net 6?
> 
> We're still in the process of planning for .NET 6, but given the state of browser support for threading and the amount of runtime work we already want to do for .NET 6 I'd say it's looking unlikely that multithreading will land for Blazor WASM in .NET 6.
> 
> > We also need .Net AOT compilation so we can draw with SkiaSharp. @mattleibow sounds like he's ready to knock that out when .Net 6 preview releases of AOT start dropping.
> 
> I do expect that WebAssembly AoT compilation will land for .NET 6, and the current thinking is that this will enable SkiaSharp scenarios. Still lots to work out though - .NET 6 planning is still a work in progress.

Oh oh Daniel, sounds weird. Every time multithreading is pushed away to some unknown future :-(

**Comment by danroth27** [comment]
@lewing @lambdageek Possible duplicate of https://github.com/dotnet/runtime/issues/68162?

**Comment by pavelsavara** [comment]
closing as duplicate of https://github.com/dotnet/runtime/issues/68162
